Missing template argument for inner type typedef.

Header:
```C++
template <typename T>
class Foo {
 public:
  class InnerType {};
};

typedef Foo<int>::InnerType Bar;

Bar func();
```

Generated:
```Rust
#[repr(C)]
#[derive(Debug, Copy, Clone)]
pub struct Foo<T> {
    pub _address: u8,
    pub _phantom_0: ::std::marker::PhantomData<T>,
}
#[repr(C)]
#[derive(Debug, Copy, Clone)]
pub struct Foo_InnerType<T> {
    pub _address: u8,
    pub _phantom_0: ::std::marker::PhantomData<T>,
}
pub use self::Foo_InnerType as Bar;
extern "C" {
    #[link_name = "_Z4funcv"]
    pub fn func() -> Bar;
}
```

Compile error
```
error[E0243]: wrong number of type arguments: expected 1, found 0
  --> generated.rs:18:22
   |
18 |     pub fn func() -> Bar;
   |                      ^^^ expected 1 type argument
```

I think it should have generated

```Rust
pub type Bar = self::Foo_InnerType<::std::os::raw::c_int>;
```

--- a/tests/expectations/tests/inner-typedef-gh422.rs
+++ b/tests/expectations/tests/inner-typedef-gh422.rs
@@ -1,0 +1,37 @@
+#![allow(
+    dead_code,
+    non_snake_case,
+    non_camel_case_types,
+    non_upper_case_globals
+)]
+
+#[repr(C)]
+#[derive(Debug, Default, Copy, Clone)]
+pub struct Foo {
+    pub _address: u8,
+}
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct Foo_InnerType<T> {
+    pub t: T,
+    pub _phantom_0: ::std::marker::PhantomData<::std::cell::UnsafeCell<T>>,
+}
+impl<T> Default for Foo_InnerType<T> {
+    fn default() -> Self {
+        let mut s = ::std::mem::MaybeUninit::<Self>::uninit();
+        unsafe {
+            ::std::ptr::write_bytes(s.as_mut_ptr(), 0, 1);
+            s.assume_init()
+        }
+    }
+}
+pub type Bar = InnerType;
+extern "C" {
+    #[link_name = "\u{1}_Z4funcv"]
+    pub fn func() -> Bar;
+}
+#[repr(C)]
+#[derive(Debug, Default, Copy, Clone)]
+pub struct InnerType {
+    pub _address: u8,
+}

--- a/tests/headers/inner-typedef-gh422.hpp
+++ b/tests/headers/inner-typedef-gh422.hpp
@@ -1,0 +1,11 @@
+template <typename T>
+class Foo {
+public:
+    class InnerType {
+        T t;
+    };
+};
+
+typedef Foo<int>::InnerType Bar;
+
+Bar func();

